### ISSUE TITLE
Question about the gt_count in MApMetric

### ISSUE THREAD

**Issue by BunnyShan** [zhreshold/mxnet-ssd#97]
@zhreshold I have noticed that the gt_count in MApMetric is calculated by all cids in preds. But what if there are some classes not preded in the class? I have tested it on VOC val set and the gt_count will be less than real ground truth num in labels some time. Is this a problem or I misunderstand this? Thank you!

**Comment by zhreshold** [comment]
You are correct, fixed now.

**Comment by BunnyShan** [comment]
@zhreshold Another question: Why SSD's input need to be resized to a predefine shape? SSD is a fcn framework, will no resize operation increases the performance?

**Comment by zhreshold** [comment]
No-resizing is potentially good for performance, but training is more complicated since you can only do batch-1 training, and there's not a easy solution for mini-batching like fast-rcnn.

**Comment by BunnyShan** [comment]
I see. In yolo2, it uses the same size in one batch and change another size after a few batches. I will look for some other solutions. Thank you!

**Comment by BunnyShan** [comment]
@zhreshold I got about mAP:65% after update the new code. Is there something wrong?

**Comment by zhreshold** [comment]
I'm verifying this.

**Comment by BunnyShan** [comment]
Thanks a lot. Fix gt_count will cause recall decrease. I will test it too.

**Comment by BunnyShan** [comment]
The gt_count fix only cause about 1% mAP drop.

**Comment by BunnyShan** [comment]
@zhreshold Is there any progress?

**Comment by zhreshold** [comment]
seems like it was related to some parameter change including lr_mult and learning rate. I did modified them for sake of a lot of GPUs, and it turns out to be not as good as the original ones.

**Comment by BunnyShan** [comment]
@zhreshold why update rescale_grad like this `'rescale_grad': 1.0 / len(ctx) if len(ctx) > 0 else 1.0` ? And previous `'rescale_grad': 1.0` makes my model don't converge.

**Comment by zhreshold** [comment]
Using multi devices will reduce the effective batch-size, 'valid' normalization in softmaxOutput and makeloss will be affected.
